### PR TITLE
Update bridge (#87)

### PR DESCRIPTION
**Pull request by rosequ**
- removed the old SM model code
- added support for newer implementation of SM model
- updated SM model code to to be compatible with the latest torchtext version.
@tuzhucheng @Impavidity please review

### CODE REVIEW

File: sm_cnn/README.md [hunk 1]
```diff
@@ -1,80 +1,149 @@
...
 
 
 ### Requirements
-gensim==1.0.1
+```
 nltk==3.2.2
 numpy==1.11.3
-pandas==0.19.2
-torch==0.1.11+b13b701
+pytorch==0.1.12
```
Comment (tuzhucheng) [inline sm_cnn/README.md]: These are outdated and can be removed, since there is already a `requirements.txt` file.

File: sm_cnn/requirements.txt [hunk 2]
```diff
@@ -1,5 +1,4 @@
-gensim==1.0.1
 nltk==3.2.1
 numpy==1.11.3
-pandas==0.19.2
+gensim==1.0.1
 pytorch==0.1.12
```
Comment (tuzhucheng) [inline sm_cnn/requirements.txt]: This is out dated, please update.

File: sm_cnn/trec_dataset.py [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
-import os
 
 class TrecDataset(data.TabularDataset):
     dirname = 'data'
     @classmethod
 
-    def splits(cls, question_id, question_field, answer_field, external_field, label_field,
-               train='train.tsv', validation='dev.tsv', test='test.tsv'):
+    def splits(cls, question_id, question_field, answer_field, external_field, label_field, root='.data',
+               train='trecqa.train.tsv', validation='trecqa.dev.tsv', test='trecqa.test.tsv'):
```
Comment (tuzhucheng) [inline sm_cnn/trec_dataset.py]: Seems like you updated to torchtext 0.2.0? If so, please remove instructions to check out a particular torchtext commit from the README and put torchtext 0.2.0 in `requirements.txt`.

File: sm_cnn/.gitignore [hunk 4]
```diff
@@ -1,5 +1,6 @@
 *pyc
-trec_eval-8.0/trec_eval
-data/
+*.pt
+text/
 trained_models/
 trec_eval-8.0/trec_eval.dSYM
```
Comment (tuzhucheng) [inline sm_cnn/.gitignore]: Remove trec_eval-8.0 stuff completely from gitignore.

File: sm_cnn/README.md [hunk 5]
```diff
@@ -1,80 +1,149 @@
...
+#use this commit number
+git reset --hard 2980f1bc39ba6af332c5c2783da8bee109796d4c
+python setup.py install
+```
 
-1. Setup repository layout.
+We use `trec_eval` for evaluation:
 
+```bash
+cd eval
```
Comment (tuzhucheng) [inline sm_cnn/README.md]: I suggest we consolidate all trec_eval utils - check out this utility function for getting `map` and `mrr`: https://github.com/castorini/Castor/blob/master/utils/relevancy_metrics.py#L6. Check here for a sample usage: https://github.com/castorini/Castor/blob/61c8c0e622caa69e8579174dee429901128dc7e1/nce/NCE-Pairwise-SM/main.py#L73.

File: sm_cnn/requirements.txt [hunk 6]
```diff
@@ -1,5 +1,4 @@
-gensim==1.0.1
-nltk==3.2.1
-numpy==1.11.3
-pandas==0.19.2
-pytorch==0.1.12
+nltk==3.2.4
+numpy==1.13.1
+pytorch==0.2.0+5f864ca
```
Comment (tuzhucheng) [inline sm_cnn/requirements.txt]: please remove the commit number from the end

Comment (rosequ) [reply sm_cnn/requirements.txt]: @tuzhucheng I think the tutorial and `nn-architecture.png` are already deleted, right?

**Comment by Impavidity** [comment]
LGTM

**Pull request merged**